eng/packaging.targets

```diff
@@ -42,4 +54,76 @@
     <MSBuild Projects="$(PkgProjPath)"
              Targets="Build" />
   </Target>
+    
+  <Target Name="AddRefAssemblies" Condition="'$(IncludeRefInPackage)' == 'true' and '$(TargetsAnyOS)' == 'true' and '@(ResolvedMatchingContract)' != ''"> 
```

Comment:
Does the condition imply that to include a ref assembly in a package, the source project needs a rid-less configuration? Unsure if that's the case everywhere currently and if that needs to be a requirement.

Reply:
> Does the condition imply that to include a ref assembly in a package, the source project needs a rid-less configuration?

i think we did  a sweep to check that all our projects has a ridless tfm which could apply on for any new and unsupported tfms.

Reply:
Right I did so recently but not for all projects and without any validation in place as reasons were different ones. What's the reason that this only applies to rid agnostic assemblies?

Reply:
Can you please change the property's name from `IncludeRefInPackage` to `PackageReferenceAssembly`? There's already the `ProduceReferenceAssembly` property in the SDK which is why I would prefer the other name.

Reply:
> Right I did so recently but not for all projects and without any validation in place as reasons were different ones

package validation will do that for you. 

> What's the reason that this only applies to rid agnostic TFMs

i originally added this to avoid removing the "custom duplicates"  from the itemgroups. we end up adding this https://github.com/dotnet/runtime/pull/54250/files/e4bae918653347e44050345f4ff8af9de1ab1f86#diff-bde310e498052e67ae73f9f1014b3ee8e067170b61c410b6b3e180fb79ed7278R113 

i can add a similar thing for symbols files if we want to remove targetsAnyOS

Reply:
> package validation will do that for you.

How can package validation validate that a rid-less source configuration is available?